### PR TITLE
[NFCI][SYCL] Support multi_ptr in convertToOpenCLType

### PR DESCRIPTION
**Pull request by aelovikov-intel**

### CODE REVIEW

File: sycl/include/sycl/sub_group.hpp [hunk 1]
```diff
@@ -42,6 +42,23 @@ namespace sub_group {
 template <typename T>
 using SelectBlockT = select_cl_scalar_integral_unsigned_t<T>;
 
+template <typename MultiPtrTy> auto convertToBlockPtr(MultiPtrTy MultiPtr) {
+  static_assert(is_multi_ptr_v<MultiPtrTy>);
+  auto DecoratedPtr = convertToOpenCLType(MultiPtr);
+  using DecoratedPtrTy = decltype(DecoratedPtr);
+  using ElemTy = remove_decoration_t<std::remove_pointer_t<DecoratedPtrTy>>;
+
+  using TargetElemTy = SelectBlockT<ElemTy>;
```
Comment (againull) [inline sycl/include/sycl/sub_group.hpp:51]: If ElemTy has cv qualifier, it looks like SelectBlockT is going to drop that qualifier.
Maybe my understanding is wrong.

Comment (aelovikov-intel) [reply sycl/include/sycl/sub_group.hpp:51]: True, but I don't think it's a regression caused by this PR.

Comment (againull) [reply sycl/include/sycl/sub_group.hpp:51]: I see. 
NIT: In that case it will be nice to have some todo maybe, otherwise I guess it will be forgotten.

**Review by againull — APPROVED** [review]

**Comment by aelovikov-intel** [comment]
Pre-commit CI failure on Win:
```
Failed Tests (1):
  SYCL :: Graph/RecordReplay/barrier_with_work.cpp
```

```
FAIL: SYCL :: Graph/RecordReplay/barrier_with_work.cpp (951 of 1890)
******************** TEST 'SYCL :: Graph/RecordReplay/barrier_with_work.cpp' FAILED ********************
Exit Code: 3221226505

Command Output (stdout):
--
# RUN: at line 1
D:/github/_work/llvm/llvm/install/bin/clang++.exe   -fsycl -fsycl-targets=spir[64](https://github.com/intel/llvm/actions/runs/7892499647/job/21541937410#step:12:65) D:\github\_work\llvm\llvm\llvm\sycl\test-e2e\Graph\RecordReplay\barrier_with_work.cpp -o D:\github\_work\llvm\llvm\build-e2e\Graph\RecordReplay\Output\barrier_with_work.cpp.tmp.out
# executed command: D:/github/_work/llvm/llvm/install/bin/clang++.exe -fsycl -fsycl-targets=spir64 'D:\github\_work\llvm\llvm\llvm\sycl\test-e2e\Graph\RecordReplay\barrier_with_work.cpp' -o 'D:\github\_work\llvm\llvm\build-e2e\Graph\RecordReplay\Output\barrier_with_work.cpp.tmp.out'
# note: command had no output on stdout or stderr
# RUN: at line 2
env ONEAPI_DEVICE_SELECTOR=level_zero:gpu  D:\github\_work\llvm\llvm\build-e2e\Graph\RecordReplay\Output\barrier_with_work.cpp.tmp.out
# executed command: env ONEAPI_DEVICE_SELECTOR=level_zero:gpu 'D:\github\_work\llvm\llvm\build-e2e\Graph\RecordReplay\Output\barrier_with_work.cpp.tmp.out'
# .---command stdout------------
# | Unexpected value at index [71](https://github.com/intel/llvm/actions/runs/7892499647/job/21541937410#step:12:72)2 for DataA: 713 (got) vs 718 (expected)
# `-----------------------------
# .---command stderr------------
# | Assertion failed: check_value(i, ReferenceA[i], DataA[i], "DataA"), file D:/github/_work/llvm/llvm/llvm/sycl/test-e2e/Graph/RecordReplay/barrier_with_work.cpp, line 118
# `-----------------------------
# error: command failed with exit status: 0xc0000409
```

Github search (https://github.com/search?q=repo%3Aintel%2Fllvm+Graph%2FRecordReplay%2Fbarrier_with_work.cpp&type=pullrequests) didn't find anything but it doesn't look like it's related to this PR either. Tagging @intel/sycl-graphs-reviewers for awareness.

I believe it's very unlikely this has been caused by this PR so I'm going to merge it and monitor post-commit afterwards. If it fails, I'll revert. I don't think it would be beneficial to just restart pre-commit and increase the load on CI.

**Pull request merged**